views/pages/GraphPage.jsx

```diff
@@ -1,60 +1,37 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { useOutletContext, useParams } from 'react-router-dom'
-import FileSaver from 'file-saver'
 import {
   Box,
-  Button,
-  DialogActions,
-  DialogContent,
-  Dialog,
+  Divider
 } from '@mui/material'
-import { Save, Functions } from '@mui/icons-material'
 
 import PageHeader from '../components/PageHeader'
 import SelectConfigurationForm from '../components/SelectConfigurationForm'
-import Matrix from '../components/Matrix.d3'
-import GraphPageTable from '../components/GraphPageTable'
+import { Graph } from '../components/Graph'
 import api from '../api'
 
-const cardSize = 20
-
 const GraphPage = () => {
   const { configurations, user, theme, setUser, setNotification } =
     useOutletContext()
-  const el = React.useRef()
-  const canvasRef = React.useRef()
-  const graphRef = React.createRef()
   const { study, subject } = useParams()
-  const [graphRendered, setGraphRendered] = React.useState(0)
-  const [graph, setGraph] = React.useState({
-    configurations: [],
-    consentDate: '',
-    matrixData: [],
-  })
-  const [graphDimensions, setGraphDimensions] = React.useState({
-    height: window.innerHeight,
-    width: window.innerWidth,
-  })
-  const [dayData, setDayData] = React.useState({
-    startFromTheLastDay: false,
-    startDay: 1,
-    lastDay: null,
-    maxDay: 1,
-  })
-  const [openStat, setOpenStat] = React.useState(false)
-  const downloadPng = () =>
-    canvasRef.current.toBlob((blob) => {
-      FileSaver.saveAs(blob, `${subject}.png`)
-    })
+  const [participants, setParticipants] = useState([])
+  useEffect(() => {
+    const asyncEffect = async () => {
```

Comment:
what do you think about moving this function out of the use effect and naming it something like fetchParticipants..?

Reply:
this is a common pattern for functions that are only called inside the useEffect. If this was moved outside the useEffect then the useEffect should technically depend on it, which would in turn mean using something like a useCallback for this function but it's just easier keeping it here 👍

Reply:
Thanks for sharing that @mikestone14